scripts/generators/pnext_chain_extraction_generator.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/python3 -i
+#
+# Copyright (c) 2015-2023 The Khronos Group Inc.
+# Copyright (c) 2015-2023 Valve Corporation
+# Copyright (c) 2015-2023 LunarG, Inc.
+# Copyright (c) 2015-2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import os
+from generators.generator_utils import (incIndent, decIndent, addIndent)
+from generators.vulkan_object import (Member)
+from generators.base_generator import BaseGenerator
+
+class PnextChainExtractionGenerator(BaseGenerator):
+    def __init__(self):
+        BaseGenerator.__init__(self)
+
+        # List of Vulkan structures that a pnext chain extraction function will be generated for
+        self.target_structs = [
+            'VkPhysicalDeviceImageFormatInfo2'
+            ]
+    
+    def generate(self):
+        out = []
+        out.append(f'''// *** THIS FILE IS GENERATED - DO NOT EDIT ***
+// See {os.path.basename(__file__)} for modifications
+
+/***************************************************************************
+*
+* Copyright (c) 2023 The Khronos Group Inc.
+* Copyright (c) 2023 Valve Corporation
+* Copyright (c) 2023 LunarG, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+****************************************************************************/\n\n''')
+        out.append('// NOLINTBEGIN\n\n') # Wrap for clang-tidy to ignore
+
+        if self.filename == 'pnext_chain_extraction.h':
+            out.append(self.generateHeader())
+        elif self.filename == 'pnext_chain_extraction.cpp':
+            out.append(self.generateSource())
+        else:
+            out.append(f'\nFile name {self.filename} has no code to generate\n')
+
+        out.append('\n// NOLINTEND') # Wrap for clang-tidy to ignore
+        self.write("".join(out))
+
+    def generateHeader(self):
+        out = []
+        out.append(f'''#pragma once
```

Comment:
no need for `f` string since not added a variable

```suggestion
        out.append('''#pragma once
```